Switched to using protobufs for all relevant mtr-api handlers

crleblanc opened the pull request:
Resolves GeoNet/mtr#136.

Updated the vendored version of protobuf and added tests for new protobuf format outputs.

junghao commented on mtr-api/field_type.go:
```diff
@@ -1,20 +1,50 @@
+type fieldType struct {
+}
+
+func (f *fieldType) proto(r *http.Request, h http.Header, b *bytes.Buffer) *weft.Result {
```
The pointer *fieldType has no harm here but I would suggest remove the pointer to be consistent with fieldModel and fieldDevice.

crleblanc replied:
Agreed.  I'll update the PR shortly.

junghao commented on mtr-api/routes_test.go:
```diff
@@ -858,3 +861,153 @@ func TestDataLatencyThreshold(t *testing.T) {
+
+	for i, ft := range ftr.Result {
+		if ft.TypeID != expectedTypeIDs[i] {
+			t.Errorf("expected %s got %s", expectedTypeIDs[i], ft.TypeID)
```
I think the check here might be a little bit dangerous since the query in "(f *fieldType) proto" has no "orderBy" clause.

crleblanc replied:
Yeah I wasn't 100% about that.  I thought since ftr.Result is a slice of struct pointers it should have a reliable order.  I take your point, it should be simple to use a map (as a set) instead of a slice, so the order doesn't matter.  Won't take long, I'll add a comment when it's ready to have a second look.

crleblanc commented:
Thanks for the comments @junghao, it should be ready now.  I noticed a few handlers had pointers that were unnecessary or had inconsistent names so I've cleaned that up.  I also simplified the test as per @gclitheroe's suggestion, just testing the first result.

Since it's still in a branch I've force pushed the new commits which has made the comments outdated.
